### ISSUE TITLE
Pay Elsewhere button - the button, when used, is not animated

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#53474]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.70-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5290609
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Open New expensify desktop (main user)
2. have a second user send some expenses to main user
3. as main user, Hold one of the expenses
4. select pay elsewhere button

## Expected Result:
The button should be animated the way it is specified here: https://github.com/Expensify/App/issues/48036#issuecomment-2316711777

## Actual Result:
The button is not animated. pay elsewhere jumps straight to payed expenses

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/ce93f452-a531-4ca3-b562-2367e013b166

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by trjExpensify** [comment]
@lanitochka17 specifically in the case of partial pay of the report when an expense is held?

**Comment by trjExpensify** [comment]
>@lanitochka17 specifically in the case of partial pay of the report when an expense is held?

Confirmed this doesn't require putting an expense on hold. I've simplified the steps in the OP. 

@Expensify/design do you know why we don't animate the button in the report header here? Jus' checking there isn't an intentional reason for that... 🤔

**Comment by shawnborton** [comment]
Hmm no idea, we should be animating it everywhere I would think?

**Comment by dannymcclain** [comment]
Yeah I think we agreed on implementing the animation everywhere but I don't think we've worked on adding the animation to the report header yet. [Here's the thread in Slack](https://expensify.slack.com/archives/CC7NECV4L/p1724447142877389) for reference.

**Comment by trjExpensify** [comment]
Hm, not sure about the "payment complete" magically showing up on each of the expense preview components when you click a button in the report header: 
<img width="808" alt="image" src="https://github.com/user-attachments/assets/ebc49b2c-ed03-4327-8628-b48cc24d02e4">

**Comment by dannymcclain** [comment]
Yeah I don't think that's the expected behavior. We essentially want the animation that shows on the _report preview_ to be mimicked in the report _header_. I don't think anything needs to be animated on the expense previews.

**Comment by shawnborton** [comment]
Yup, definitely agree with that.

**Comment by trjExpensify** [comment]
Yeah, this just looks a bit strange to me when the button disappears. I'll defer to the design team on it though. 

<img width="1511" alt="image" src="https://github.com/user-attachments/assets/c2cd47b0-018f-4eaf-be26-d0874b54bbf2">

**Comment by dannymcclain** [comment]
I agree with Tom, I think that does look weird. I think losing the container like that is part of it... Would it be crazy to do something like this where we animate from the pay text to the payment completed text (loader in between if we need it) and then after the payment complete just animate the whole button out?

![CleanShot 2024-12-04 at 14 12 15@2x](https://github.com/user-attachments/assets/1e581506-0c7b-4a3d-90df-02b622572532)

Is that going to get tricky with button widths? 🤔 Going to wait and see what the other designers have to say.

**Comment by dubielzyk-expensify** [comment]
I like changing it to what you're suggesting Danny, but I think we should do this for the preview as well?

**Comment by dannymcclain** [comment]
>I like changing it to what you're suggesting Danny, but I think we should do this for the preview as well?

Oh I'm not against that. That would make them ultra-consistent which could be nice. Let's see what @shawnborton has to say!

**Comment by shawnborton** [comment]
Yup, I also like that suggestion - let's do it!

**Comment by trjExpensify** [comment]
Let us know what you think, @ntdiary!

**Comment by dannymcclain** [comment]
That video looks like the correct behavior to me, though we may want to tweak the "exit" animation a tad. Would love to get the @Expensify/design team's thoughts on that.

**Comment by shawnborton** [comment]
Nice, that does look good. I suppose it would be interesting to see a revised exist animation that just does some kind of scaleOut, where we animate from a scale of 1 to a scale of 0? That might give us a similar effect as shrinking the height, and it might feel more natural in the header. 

That being said, I think it makes sense to keep the animation the same everywhere and not try to have a custom version just for the header. So perhaps we can try it everywhere and see how it goes?

**Comment by dubielzyk-expensify** [comment]
Agree with that Shawn. I'd be keen to try it. Sometimes the 1 to 0 scale can be a bit too much, but we should try it. 

Alternatively we can do scale 1->0.5 and just do opacity 1->0. But lets start with the scale first.

**Comment by trjExpensify** [comment]
@ntdiary what do you think?

**Comment by trjExpensify** [comment]
>If it still needs improvements, I suggest discussing them in the PR

Yep, I think [this is the style](https://github.com/Expensify/App/issues/53474#issuecomment-2518468709) we landed on.

**Comment by lakchote** [comment]
@ntdiary I agree with you, we should be `DRY` and prevent code duplication with the animations boilerplate code if it's going to be reused.

I'm also in favour of creating a test case for it.

@abzokhattab when you raise the PR please assign `@Expensify/design` team for their final approval.

**Comment by trjExpensify** [comment]
PR has been reviewed. @abzokhattab can you address the comments, please? Thanks!

**Comment by trjExpensify** [comment]
@abzokhattab can you give us an ETA on addressing the comments?

**Comment by trjExpensify** [comment]
Bumped the PR, let's get this done, it has been open for a month.

**Comment by lakchote** [comment]
This issue has been ongoing for a while.

@abzokhattab please address @ntdiary comments [here](https://github.com/Expensify/App/pull/54490#issuecomment-2643720141)

**Comment by abzokhattab** [comment]
Added two comments please have a look and let me know what u think @ntdiary

**Comment by trjExpensify** [comment]
Latest on the PR [here](https://github.com/Expensify/App/pull/54490#issuecomment-2666125350), Melv. Maybe take it to slack @ntdiary for more collab and eyes if you're stuck?

**Comment by lakchote** [comment]
PR has been merged.

**Comment by trjExpensify** [comment]
👋 let's get the checklist in for tomorrow!

**Comment by trjExpensify** [comment]
Payment summary as follows: 

- $250 to @ntdiary for C+ (go ahead and request)
- $250 to @abzokhattab for the PR (paid!)

**Comment by JmillsExpensify** [comment]
$250 approved for @ntdiary

**Comment by trjExpensify** [comment]
Closing!
